### PR TITLE
Adding samples library

### PR DESCRIPTION
**Pull request by flossypurse**
## What does this PR do?

Adds a new page that is a complete library of our samples.

Preview: https://deploy-preview-487--mystifying-fermi-1bc096.netlify.app/docs/samples-library

Adds big link to it here: https://deploy-preview-487--mystifying-fermi-1bc096.netlify.app/application-development

## Notes to reviewers

This page pulls in the directories for the samples repo's READMEs.

Some work will be done to those READMEs so that there is a consistent representation.

### CODE REVIEW

File: docs/samples-library.md [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+  Works in conjunction with the ["Hello World!" from scratch tutorial in Java](/docs/go/hello-world-tutorial).
+
+- [**Money transfer application template in Go**](https://github.com/temporalio/money-transfer-project-template-go): Provides a quick start development app for users.
+  It demonstrates a basic money transfer Workflow Definition and works in conjunction with the [Run your first app tutorial in Go](/docs/go/run-your-first-app-tutorial).
+
+- [**Subscription style Workflow Definition in Go**](https://github.com/temporalio/subscription-workflow-project-template-go): Demonstrates some of the patterns that could be implemented for a subscription style business process.
+
+- [**eCommerce application example in Go**](https://github.com/temporalio/temporal-ecommerce): Showcases a per user shopping cart style Workflow Definition that includes an API for adding and removing items from the cart as well as a web UI.
+  This application sample works in conjunction with the [eCommerce in Go tutorial](/blog/tags/go-ecommerce-tutorial).
+
```
Comment (djmagee) [inline docs/samples-library.md:23]: Perhaps add a para such as the following (either here or in the source)?

The following sections describe the available Go samples and how to run them.

File: docs/samples-library.md [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+  It demonstrates a basic money transfer Workflow Definition and works in conjunction with the [Run your first app tutorial in Java](/docs/java/run-your-first-app-tutorial).
+
+- [**Subscription style Workflow Definition in Java**](https://github.com/temporalio/subscription-workflow-project-template-java): Demonstrates some of the patterns that could be implemented for a subscription style business process.
+
+<!--SNIPSTART samples-java-readme-samples-directory {"enable_source_link": false, "enable_code_block": false}-->
+<!--SNIPEND-->
+
+## PHP
+
+- [**Subscription style Workflow Definition in PHP**](https://github.com/temporalio/subscription-workflow-project-template-php): Demonstrates some of the patterns that could be implemented for a subscription style business process.
```
Comment (djmagee) [inline docs/samples-library.md:42]:
```suggestion
- [**Subscription-style Workflow Definition in PHP**](https://github.com/temporalio/subscription-workflow-project-template-php): Demonstrates some of the patterns that could be implemented for a subscription-style business process.
```

**Review by djmagee — APPROVED** [review]

File: docs/samples-library.md [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+- [**Polyglot application**](https://github.com/temporalio/temporal-polyglot): Showcases how Workflow Executions, that are written in different languages, can send messages to each other.
+  Go, Java, PHP, and NodeJS SDKs are represented in this sample.
+  It also shows how to properly propagate errors including how to do it across Workflows written in different languages.
+
+## Go
+
+- [**Hello world application template in Go**](https://github.com/temporalio/hello-world-project-template-go): Provides a quick start development app for users.
+  Works in conjunction with the ["Hello World!" from scratch tutorial in Java](/docs/go/hello-world-tutorial).
+
+- [**Money transfer application template in Go**](https://github.com/temporalio/money-transfer-project-template-go): Provides a quick start development app for users.
```
[the remark this answers is not in the document]
Comment (flossypurse) [reply docs/samples-library.md:16]: Well, the actual name of the repo and application is "Money transfer", so not sure about adding the hyphens on this one. But yes on quick-start.

[the remark this answers is not in the document]
Comment (djmagee) [reply docs/samples-library.md:16]: Yeah, I can argue this one both ways.

File: docs/samples-library.md (hunk 1 above)
Comment (flossypurse) [reply docs/samples-library.md:23]: Going to revise those READMEs so that the format is the same.

**Pull request merged**